code/controllers/subsystem/chat_pings.dm

```diff
@@ -0,0 +1,17 @@
+SUBSYSTEM_DEF(chat_pings)
+	name = "Chat Pings"
+	flags = SS_NO_INIT
+	runlevels = RUNLEVEL_INIT | RUNLEVEL_LOBBY | RUNLEVEL_SETUP | RUNLEVEL_GAME | RUNLEVEL_POSTGAME // ALL OF THEM
+	wait = 30 SECONDS // Chat pings every 30 seconds
+	/// List of all held chat datums
+	var/list/datum/chatOutput/chat_datums = list() // Do NOT put this in Initialize(). You will cause issues.
+
+/datum/controller/subsystem/chat_pings/fire(resumed)
+	for(var/C in chat_datums)
+		var/datum/chatOutput/CO = C
```

Comment:
```suggestion
	for(var/datum/chatOutput/CO as anything in chat_datums)
```
smh do your typeless loops more efficiently

Reply:
It looks less archaic. Ill take it.

Reply:
Isn't the standard still to typecast it afterwards? #15729 at least seemed to indicate that there wasn't a noticable performance impact between the two.

Reply:
The problem in that issue was changing the entire codebase for a minimal boost. This is just in one place and looks less stupid than iterating as a blank var then typecasting after.